owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/details/FileDetailsFragment.kt

```diff
@@ -0,0 +1,449 @@
+
+        val message = if (workInfo.isDownload()) {
+            getString(R.string.downloader_download_failed_ticker)
+        } else { // Transfer is upload (?)
+            getString(R.string.uploader_upload_failed_ticker)
+        }
+        showMessageInSnackbar(message)
+    }
+
+    private fun updateLayoutForCancelledTransfer(workInfo: WorkInfo) {
+        showProgressView(isTransferGoingOn = false)
+
+        val message = if (workInfo.isDownload()) {
+            getString(R.string.downloader_download_canceled_ticker)
+        } else { // Transfer is upload (?)
+            getString(R.string.uploader_upload_canceled_ticker)
+        }
+        showMessageInSnackbar(message)
+        fileDetailsViewModel.updateActionInDetailsView(NONE)
+    }
+
+    /**
+     * Show or hide progress for transfers.
+     */
+    private fun showProgressView(isTransferGoingOn: Boolean) {
+        binding.fdProgressBar.isVisible = isTransferGoingOn
+        binding.fdProgressText.isVisible = isTransferGoingOn
+        binding.fdCancelBtn.isVisible = isTransferGoingOn
+
+        // Invalidate to reset the menu items -> Show/Hide Download/Sync/Cancel
+        requireActivity().invalidateOptionsMenu()
+    }
+
+    // TODO: Move navigation to a common place.
+    private fun navigateToPreviewOrOpenFile(fileWaitingToPreview: OCFile) {
+        val fileDisplayActivity = requireActivity() as FileDisplayActivity
+        when {
+            PreviewImageFragment.canBePreviewed(fileWaitingToPreview) -> {
+                fileDisplayActivity.startImagePreview(fileWaitingToPreview)
+            }
+            PreviewAudioFragment.canBePreviewed(fileWaitingToPreview) -> {
+                fileDisplayActivity.startAudioPreview(fileWaitingToPreview, 0)
+            }
+            PreviewVideoFragment.canBePreviewed(fileWaitingToPreview) -> {
+                fileDisplayActivity.startVideoPreview(fileWaitingToPreview, 0)
+            }
+            PreviewTextFragment.canBePreviewed(fileWaitingToPreview) -> {
+                fileDisplayActivity.startTextPreview(fileWaitingToPreview)
+            }
+            else -> fileDisplayActivity.openOCFile(fileWaitingToPreview)
+        }
+    }
+
+    override fun updateViewForSyncInProgress() {
+        TODO("Not yet implemented")
+    }
+
+    override fun updateViewForSyncOff() {
+        TODO("Not yet implemented")
+    }
+
+    override fun onFileMetadataChanged(updatedFile: OCFile?) {
+        // Nothing to do here. We are observing the oCFile from database, so it should be refreshed automatically
+    }
+
+    override fun onFileMetadataChanged() {
+        TODO("Not yet implemented")
+    }
+
+    override fun onFileContentChanged() {
+        TODO("Not yet implemented")
+    }
+
+    companion object {
+        private const val ARG_FILE = "FILE"
+        private const val ARG_ACCOUNT = "ACCOUNT"
+        private const val ARG_SYNC_FILE_AT_OPEN = "ARG_SYNC_FILE_AT_OPEN"
```

Comment:
Following the previous variables pattern, the value of this one should be `"SYNC_FILE_AT_OPEN"`, but it's ok if you don't want to change it since it is internal